### PR TITLE
Port for RT-Thread

### PR DESCRIPTION
**Pull request by WKJay**
https://www.rt-thread.io
https://www.rt-thread.org

### CODE REVIEW

**Comment by dgarske** [comment]
Hi @WKJay , thanks for the RTOS port to rt-thread.org. In order to accept code contributions we need a signed contributor agreement. Can you please email support@wolfssl.com and reference this PR along with your contact information? Okay to test.

Thanks,
David Garske, wolfSSL

**Comment by WKJay** [comment]
Hi @dgarske 

I have sent the email, please check.

Thanks,
Jay

**Review by tmael — COMMENTED** [review]
Looks good. Can you add the new files in wolfssl/IDE/include.am?

File: IDE/ECLIPSE/RTTHREAD/README.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+ED25519  test passed!
+PKCS7encrypted  test passed!
+PKCS7signed     test passed!
+PKCS7enveloped  test passed!
+PKCS7authenveloped  test passed!
+logging  test passed!
+mutex    test passed!
+memcb    test passed!
+```
+### `WOLFSSL_BENCHMARK_TEST` output of benchmark_test()
```
Comment (dgarske) [inline IDE/ECLIPSE/RTTHREAD/README.md:117]: Can you include the hardware and compiler / optimization level used for this benchmark?

File: IDE/ECLIPSE/RTTHREAD/user_settings.h [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+#define USE_TEST_GENSEED
+#define NO_DEV_RANDOM
+#define HAVE_PKCS7
+#define HAVE_AES_KEYWRAP
+#define HAVE_X963_KDF
+#define WOLFSSL_AES_DIRECT
+/* adjust CURRENT_UNIX_TS to seconds since Jan 01 1970. (UTC)
+You can get the current time from https://www.unixtimestamp.com/
+*/
+#define CURRENT_UNIX_TS 1542605837
```
Comment (dgarske) [inline IDE/ECLIPSE/RTTHREAD/user_settings.h:45]: Should this be `1542605837UL`?

File: IDE/ECLIPSE/RTTHREAD/wolfssl_test.c [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+#include <stdint.h>
+#include <wolfcrypt/test/test.h>
+#include <wolfcrypt/benchmark/benchmark.h>
+
+int wolfssl_test(void){
+     wolfcrypt_test(NULL);
```
Comment (dgarske) [inline IDE/ECLIPSE/RTTHREAD/wolfssl_test.c:26]: Consider adding `#ifdef` for `NO_CRYPT_TEST` and `NO_CRYPT_BENCHMARK`.

**Review by dgarske — CHANGES_REQUESTED** [review]
Please add the new files as `EXTRA_DIST` to `include.am`. You can create one in `IDE/ECLIPSE/RTTHREAD`. See other include.am files for template. Then add it to the `IDE/include.am`.

**Comment by WKJay** [comment]
@dgarske  I've updated the PR, please check. Thanks

**Review by tmael — APPROVED** [review]

**Review by dgarske — APPROVED** [review]
Thanks! Over to Todd for final review

**Review by toddouska — APPROVED** [review]

**Pull request merged**